nuget/readme.md

```diff
@@ -2,6 +2,72 @@
 
 ## Overview
 
-Please visit [Microsoft.Windows.CsWinRT](https://www.nuget.org/packages/Microsoft.Windows.CsWinRT/) for official Microsoft-signed builds of the NuGet package. 
+Please visit [Microsoft.Windows.CsWinRT](https://www.nuget.org/packages/Microsoft.Windows.CsWinRT/) for official Microsoft-signed builds of the NuGet package.
+
+The Microsoft.Windows.CsWinRT NuGet package provides WinRT projection support for the C# language.  Adding a reference to this package causes C# projection sources to be generated for any Windows metadata referenced by the project.  For details on package use, please consult the Microsoft.Windows.CsWinRT github repo's [readme.md](https://github.com/microsoft/cswinrt/blob/master/README.md).
+
+C#/WinRT detects Windows metadata from:
+* Platform winmd files in the SDK 
+* NuGet package references containing winmd files
+* Other project references producing winmd files
+* Raw winmd file references
+
+For any winmd file discovered above, C#/WinRT creates reference (consuming) projection sources (*.cs files).  These sources can then be compiled into a projection assembly for client projects to reference, or compiled directly into an app.
+
+## Details
+
+C#/WinRT sets the following C# compiler properties:
+
+| Property | Value | Description |
+|-|-|-|
+| AllowUnsafeBlocks | true | Permits compilation of generated unsafe C# blocks |
+
+## Customizing
+
+C#/WinRT behavior can be customized with these project properties:
+
+| Property | Value(s) | Description |
+|-|-|-|
+| CsWinRTEnabled | true \| *false | Enables projection-related targets, defaults to true if CsWinRTFilters or CsWinRTParams are defined |
+| CsWinRTMessageImportance | low \| *normal \| high | Sets the [importance](https://docs.microsoft.com/en-us/visualstudio/msbuild/message-task?view=vs-2017) of C++/WinRT build messages (see below) |
```

Comment:
Should this say C#/WinRT build messages? Instead of C++/WinRT ?

Reply:
thanks!